src/main/scala/example/Task1RepartitioningApproach.scala

```diff
@@ -0,0 +1,85 @@
+package example
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.functions.col
+
+object Task1RepartitioningApproach {
+  // QUERY: Find spendings (both already paid or not) of different users on products from different categories.
+
+  def transformationTask1WithRepartitioningApproach(
+      spark: SparkSession,
+      notCompletedOrdersDF: DataFrame,
+      completedOrdersDF: DataFrame,
+      notCompletedPaymentsDF: DataFrame,
+      completedPaymentsDF: DataFrame,
+      usersDF: DataFrame,
+      productsDF: DataFrame,
+      categoriesDF: DataFrame
+    ): DataFrame = {
+    usersDF
+      .repartition(col("UserId"))
+      .createOrReplaceTempView("users")
+    productsDF
+      .repartition(col("CategoryId"))
+      .createOrReplaceTempView("products")
+    categoriesDF
+      .repartition(col("CategoryId"))
+      .createOrReplaceTempView("categories")
+//  completedPaymentsDF
```

Comment:
Why would you share in the PR commented code?

Reply:
yeah, git is a nice history keeper ;)
So, as U've done a lot of commits you could hide the commented out code in the commits too ;)